financial/exponential_moving_average.py

```diff
@@ -0,0 +1,72 @@
+"""
+    Calculates exponential moving average (EMA) on the series of numbers
+    Wikipedia Reference: https://en.wikipedia.org/wiki/Exponential_smoothing
+    Reference: https://www.investopedia.com/terms/e/ema.asp#toc-what-is-an-exponential-moving-average-ema
+
+    Exponential moving average is used in finance to analyze changes stock prices.
+    EMA is used in conjunction with Simple moving average (SMA), EMA reacts to the
+    changes in the value quicker than SMA, which is one of the advantages of using EMA.
+"""
+
+from collections.abc import Iterator
+
+
+def exponential_moving_average(
+    series_generator: Iterator[float], window_size: int
```

Comment:
Just plain `series` is fine.
```suggestion
    series: Iterator[float], window_size: int
```